### DISCUSSION TITLE
Use Filter for select some relations in multirelation

### DISCUSSION THREAD

**Discussion by dgotskin** [pocketbase/pocketbase#4319] [Q&A] [answered]
Hello. Please help me. How to do the following, if possible:

Input data: There is a collection of photos, one of the fields of which is "tags" in the form of a many-to-many relationship with the "tags" collection.

The requirement is to retrieve all photos that contain, for example, two tags: "portrait" and "bw", in such a way that only black and white portrait photos are returned.

Currently, I can select all photos with one tag, for example, like this: **?filter(tags.slug?='portrait')** to retrieve all portrait photos.

When I try to use the filter **?filter=(tags.slug?='portrait' && tags.slug?='bw')**, I don't get any results. I've tried different approaches but couldn't figure it out...

I've been scratching my head for a while but couldn't come up with a solution. Is it possible to achieve this on the "photo" collection (base type), or do I need to create an additional type like a View?

**Comment by ganigeorgiev** [comment, marked as answer]
What you are trying to do is not supported at the moment.
Your example filter doesn't work because `tags.slug` targets a single row value and that cannot be 2 different things at the same time. Internally your filter will produce a pseudo SQL query like:

```sql
SELECT DISTINCT photos.*
FROM photos
LEFT JOIN tags on photos.tags HAS tags.id -- fake HAS operator for simplicity
WHERE tags.slug = "portrait" AND tags.slug = "bw" -- cannot be both portrait and bw
```

If you filter by the tag ids instead a workaround could be to use the `~` operator and loosely check whether the serialized `tags` json value contains the ids you are looking for. For example:

```js
// fetch the portrait and bw tags (if not already)
const tags = await pb.collection("tags").getFullList({
    filter: `slug = "portrait" || slug = "bw"`,
});

const photos = await pb.collection("photos").getFullList({
    // the same as `tags ~ '"portrait_ID"' && tags ~ '"bw_ID"'`
    filter: tags.map((t) => `tags ~ '"${t.id}"'`).join(" && ")
})
```

**Reply by dgotskin** [reply]
Yes, this worked for me. In reality, it would be more convenient to select by tag attributes, but selecting by ID is also a viable option. Thank you very much for the answer!
